ext/openssl/ossl_pkey.c

```diff
@@ -568,6 +614,25 @@ ossl_pkey_inspect(VALUE self)
                       OBJ_nid2sn(nid));
 }
 
+/*
+ *  call-seq:
+ *     key.private? => true or false
+ *
+ *  Returns whether this PKey instance has a private key. The private key can
+ *  be retrieved with PKey#private_to_{der,pem,raw}.
+ */
+static VALUE ossl_pkey_is_private(VALUE self)
+{
+    EVP_PKEY *pkey;
+    size_t len;
+
+    GetPKey(self, pkey);
+    len = EVP_PKEY_size(pkey);
+    unsigned char str[len];
+
+    return EVP_PKEY_get_raw_private_key(pkey, str, &len) == 1 ? Qtrue : Qfalse;
```

Comment:
I'm pretty sure I'm not doing this right. According to https://www.openssl.org/docs/man1.1.1/man3/EVP_PKEY_get_raw_private_key.html:

> EVP_PKEY_get_raw_private_key() fills the buffer provided by priv with raw private key data. The size of the priv buffer should be in *len on entry to the function, and on exit *len is updated with the number of bytes actually written. If the buffer priv is NULL then *len is populated with the number of bytes required to hold the key.

However if I pass NULL instead of str the tests fail 🤔